### PR TITLE
Blog App: Setup Models

### PR DESCRIPTION
**Pull request by Shisui6**
## Summary 🧾
- Create model classes for all table entities
- Set up associations between models
- Add custom methods to each model

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Post model
+class Post < ApplicationRecord
+  has_many :comments
+  belongs_to :user
```
Comment (OLIPLICHE) [inline app/models/post.rb]: - Hi, according to the ERD diagram, **Post has many likes**. I would suggest adding it.

File: app/models/post.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+  def update_posts_counter
+    User.find_by(id: user_id).increment(:postsCounter).save
+  end
```
Comment (OLIPLICHE) [inline app/models/post.rb]: - [ ] Optional:
- Suggestions: You can use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_post_counter` method every time a new post has been saved to the database, then make the `update_post_counter` method private.
- EXAMPLE:
```ruby
  after_save :update_post_counter

  private

  def update_post_counter
    author.increment!(:posts_counter)
  end
```

File: app/models/comment.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+# Comment model
+class Comment < ApplicationRecord
+  belongs_to :user
+  belongs_to :post
+
+  def update_comments_counter
+    Post.find_by(id: post_id).increment(:commentsCounter).save
+  end
```
Comment (OLIPLICHE) [inline app/models/comment.rb]: - [ ] Optional:
- Suggestions: You can use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_comments_counter` method every time a new post has been saved to the database, then make the `update_comments_counter` method private.
- EXAMPLE:
```ruby
  after_save :update_comments_counter

  private

  def update_comments_counter
      Post.find_by(id: post_id).increment(:commentsCounter).save
  end
```

File: app/models/like.rb [hunk 4]
```diff
@@ -0,0 +1,9 @@
+# Like model
+class Like < ApplicationRecord
+  belongs_to :user
+  belongs_to :post
+
+  def update_likes_counter
+    Post.find_by(id: post_id).increment(:likesCounter).save
+  end
```
Comment (OLIPLICHE) [inline app/models/like.rb]: - [ ] Optional

- **Just like we have in user and comment:**  `after_save :update_likes_counter`

**Review by OLIPLICHE — CHANGES_REQUESTED** [review]
Hi @Shisui6, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Linter is green (the last commit has a green check mark (✅)) 👏
- README file is descriptive 👏

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by uwadonat — APPROVED** [review]
## Hi @Shisui6,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
### Congratulations! 🎉 
### Highlights
- All project requirements were implemented ✔️
- No linter errors ✔️
- Professional readme file ✔️
- Correct workflow was used ✔️
- Good pull request message ✔️
thanks-awesome

# Status: Approved ✔️✔️✔️

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100%.
Please, remember to tag me @uwadonat in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**